src/types/cms.cc

```diff
@@ -0,0 +1,128 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+#include "cms.h"
+
+#include <algorithm>
+#include <cmath>
+#include <cstdint>
+#include <vector>
+
+#include "glog/logging.h"
+
+void CMSketch::CMSDimFromProb(double error, double delta, uint32_t& width, uint32_t& depth) {
+  LOG(INFO) << error << delta;
+  width = std::ceil(2 / error);
+  depth = std::ceil(std::log10(delta) / std::log10(0.5));
+}
+
+size_t CMSketch::IncrBy(const char* item, size_t item_len, size_t value) {
+  size_t min_count = std::numeric_limits<size_t>::max();
+
+  for (size_t i = 0; i < depth_; ++i) {
+    uint32_t hash = HllMurMurHash64A(item, static_cast<int>(item_len), i);
```

Comment:
I don't know getting u32 from u64 hash a good idea...

Reply:
@mapleFU  tyty for the review, i'll be revising it tmr.